### PR TITLE
Fix typos and grammar errors in docstrings and comments

### PR DESCRIPTION
**Pull request by qianyun210603**
## Description
Fix some types and missing's in doc strings.

## Motivation and Context
improve code quality

## How Has This Been Tested?

- [x] Pass the test by running: `pytest qlib/tests/test_all_pipeline.py` under upper directory of `qlib`.
- [ ] If you are adding a new feature, test on your own test scripts.

## Screenshots of Test Results (if appropriate):
1. Pipeline test:
2. Your own tests:

## Types of changes

- [ ] Fix bugs
- [ ] Add new feature
- [x] Update documentation

### CODE REVIEW

**Review by ChiahungTai — APPROVED** [review]
@qianyun210603 
Thanks to your effort.
@you-n-g 
LGTM

File: qlib/data/dataset/handler.py [hunk 1]
```diff
@@ -685,7 +685,7 @@ def cast(cls, handler: "DataHandlerLP") -> "DataHandlerLP":
             the converted processed data
         """
         new_hd: DataHandlerLP = object.__new__(DataHandlerLP)
-        new_hd.from_cast = True  # add a mark for the casted instance
+        new_hd.from_cast = True  # add a mark for the cast instance
```
Comment (you-n-g) [inline qlib/data/dataset/handler.py:688]: Why not casted?

Comment (qianyun210603) [reply qlib/data/dataset/handler.py:689]: The past tense of cast is: cast in past simple is cast. and past participle is cast. There is no `casted`.

https://engverbs.com/grammar/cast#:~:text=The%20past%20tense%20of%20cast%20is%3A%20cast%20in,cast.%20The%20past%20participle%20of%20cast%20is%20cast.

File: qlib/contrib/model/pytorch_alstm.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ class ALSTM(Model):
     d_feat : int
         input dimension for each time step
     metric: str
-        the evaluate metric used in early stop
+        the evaluated metric used in early stop
```
[the remark this answers is not in the document]
Comment (qianyun210603) [reply qlib/contrib/model/pytorch_alstm.py]: corrected

File: qlib/contrib/model/pytorch_add.py [hunk 3]
```diff
@@ -36,7 +36,7 @@ class ADD(Model):
      d_feat : int
          input dimensions for each time step
      metric : str
-         the evaluate metric used in early stop
+         the evaluated metric used in early stop
```
[the remark this answers is not in the document]
Comment (qianyun210603) [reply qlib/contrib/model/pytorch_add.py]: corrected

File: qlib/contrib/model/pytorch_adarnn.py [hunk 4]
```diff
@@ -28,7 +28,7 @@ class ADARNN(Model):
     d_feat : int
         input dimension for each time step
     metric: str
-        the evaluate metric used in early stop
+        the evaluated metric used in early stop
```
[the remark this answers is not in the document]
Comment (qianyun210603) [reply qlib/contrib/model/pytorch_adarnn.py]: corrected

File: qlib/backtest/exchange.py [hunk 5]
```diff
@@ -540,8 +540,8 @@ def generate_amount_position_from_weight_position(
         direction: OrderDir = OrderDir.BUY,
     ) -> dict:
         """
-        The generate the target position according to the weight and the cash.
-        NOTE: All the cash will assigned to the tradable stock.
+        Generates the target position according to the weight and the cash.
+        NOTE: All the cash will assign to the tradable stock.
```
[the remark this answers is not in the document]
Comment (qianyun210603) [reply qlib/backtest/exchange.py]: corrected

File: qlib/backtest/exchange.py [hunk 6]
```diff
@@ -271,7 +271,7 @@ def _get_limit_type(self, limit_threshold: Union[tuple, float, None]) -> str:
             raise NotImplementedError(f"This type of `limit_threshold` is not supported")
 
     def _update_limit(self, limit_threshold: Union[Tuple, float, None]) -> None:
-        # $close is may contains NaN, the nan indicates that the stock is not tradable at that timestamp
+        # $close is may contain NaN, the nan indicates that the stock is not tradable at that timestamp
```
[the remark this answers is not in the document]
Comment (qianyun210603) [reply qlib/backtest/exchange.py]: corrected

File: qlib/data/dataset/handler.py [hunk 7]
```diff
@@ -347,7 +347,6 @@ class DataHandlerLP(DataHandler):
         - These processors only apply to the learning phase.
 
     Tips to improve the performance of data handler
-
```
[the remark this answers is not in the document]
Comment (qianyun210603) [reply qlib/data/dataset/handler.py]: added back

File: qlib/data/dataset/handler.py (hunk 1 above)
Comment (you-n-g) [reply qlib/data/dataset/handler.py:689]: Sorry for my bad grammar...

**Comment by you-n-g** [comment]
It looks great now!
Merged.
Thanks.

**Pull request merged**